unified-runtime/scripts/strip-guarded-lines.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2025 Intel Corporation
+#
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+
+"""
+This script is a basic pre-processor which deals with conditionally excluding
+blocks of lines based on pre and post guard marker lines, a list of guard
+names to include in the output, and an input file.
+
+- Pre and post guard marker lines are always removed weather the are guards
```

Comment:
This sentence doesn't really make sense.

Reply:
What about this?

```
- Pre and post guard marker lines will always be removed even when those guard
  names have not been specified for inclusion.
```